### ISSUE TITLE
Make manifest_provider.git more efficient

### ISSUE THREAD

**Issue by mikepurvis** [ros-infrastructure/rosdistro#63]
Right now for non-github repos, it has to clone the entire thing just to grab the package.xml. Per the [comment here](https://github.com/ros-infrastructure/rosdistro/blob/39455f500c8f771939e0286672547a06e4d9c233/src/rosdistro/manifest_provider/git.py#L59), this is due to a limitation with Git 1.7.9, but the oldest supported platform now is Ubuntu Trusty, which has Git 1.9.1.

So it should be possible now to change this so that it does a shallow clone of only the commit that is the tag. It's a significant savings, particularly on long-running repos which have had a lot of churn:

```
$ git clone https://github.com/ros-gbp/navigation-release.git
Cloning into 'navigation-release'...
remote: Counting objects: 67383, done.
remote: Total 67383 (delta 0), reused 0 (delta 0), pack-reused 67383
Receiving objects: 100% (67383/67383), 8.02 MiB | 1.14 MiB/s, done.
Resolving deltas: 100% (22599/22599), done.
Checking connectivity... done.
```

```
$ git clone https://github.com/ros-gbp/navigation-release.git --depth 1 --branch upstream/1.13.1
Cloning into 'navigation-release'...
remote: Counting objects: 411, done.
remote: Compressing objects: 100% (307/307), done.
remote: Total 411 (delta 105), reused 258 (delta 66), pack-reused 0
Receiving objects: 100% (411/411), 469.52 KiB | 0 bytes/s, done.
Resolving deltas: 100% (105/105), done.
Checking connectivity... done.
```

Cloning from Github for example's sake, but obviously this issue affects primarily private repos on Bitbucket, internal Gitlab instances, etc. And I may yet submit a manifest provider for Gitlab, but just this would be a simple change that could have a pretty big positive impact.

**Comment by dirk-thomas** [comment]
Currently the Python tools like `python-rosdistro` are distribution agnostic so the same version gets released into all Ubuntu distros / Debian distros (listed in https://github.com/ros-infrastructure/rosdistro/blob/39455f500c8f771939e0286672547a06e4d9c233/stdeb.cfg#L6). Based on that such a change wouldn't be possible since it would break installations of ROS on older distributions which only have older versions of git.

That being said it might be reasonable to consider changing our policy and removing older distros form this set of suites so that future releases are only made into "current" Ubuntu / Debian distributions. @tfoote @wjwwood What do you think about this?

**Comment by dirk-thomas** [comment]
We have discussed this case and came to the following conclusion:

* once an Ubuntu distribution has been EOLed (and we have stopped building for it on the farm) we will take it out of the list of suites in the stdeb.cfg files, so newer releases are only published to the remaining list

This ensures that if there will be a breaking change in the future (as e.g. the recent migration to REP 143) all non-EOL platform will receive a new version to keep working. If users still use EOLed platforms they might not even be able to use the tools to build from source.

Since all current and future ROS distros are aligned with the Ubuntu schedule there is no problem. For the past distros which e.g. worked on `Precise` the case is that the ROS distro is already EOLed (since they are not LTS releases) but Ubuntu still maintains them. We decided to keep maintaining Precise for from-source builds until Ubuntu EOLs it.
